evaluator/evaluate.go

```diff
@@ -158,27 +174,86 @@ func evaluateWithVmaas(ctx context.Context, tx *gorm.DB, updatesReq *vmaas.Updat
 		evaluationCnt.WithLabelValues("error-database-commit").Inc()
 		return nil, errors.New("database commit failed")
 	}
-	return vmaasData, nil
+	return updatesData, nil
 }
 
-func tryGetVmaasRequest(tx *gorm.DB, accountID int, inventoryID string,
-	requested *base.Rfc3339Timestamp) (*vmaas.UpdatesV3Request, *models.SystemPlatform, error) {
-	system := tryGetSystem(tx, accountID, inventoryID, requested)
-	if system == nil {
-		return nil, nil, nil
+func getUpdatesData(ctx context.Context, tx *gorm.DB, system *models.SystemPlatform) (*vmaas.UpdatesV2Response, error) {
+	if !enableYumUpdatesEval {
+		return getVmaasUpdates(ctx, tx, system)
+	}
+
+	var vmaasErr error
+	var yumErr error
+	vmaasData, vmaasErr := getVmaasUpdates(ctx, tx, system)
+	yumUpdates, yumErr := tryGetYumUpdates(system)
+	if vmaasErr != nil && yumErr != nil {
+		return nil, errors.Wrap(vmaasErr, yumErr.Error())
+	}
+	if vmaasErr != nil || vmaasData == nil {
+		return yumUpdates, nil
+	}
+	if yumErr != nil || yumUpdates == nil {
+		return vmaasData, nil
 	}
+	if yumUpdates == nil && vmaasData == nil {
+		return nil, nil
+	}
```

Comment:
Having no yum updates is not an (hard) error. It may happen on RHEL6 which does not support it or in case there's and old insights client on the system.

Reply:
I think we only skip evaluation with VMaaS, no hard errors.

Reply:
If both are nil